pidswallow

```diff
@@ -65,59 +72,60 @@ swallow() {
         # 3 -> window lacks a _NET_WM_PID property.
         # 4 -> window already swallowed
         # 5 -> swallow_command failed
-        unset cwid pwid cname pname ppid cpid # similar to local.
+        unset cwid pwid cname pname ppid cpid x y w h # similar to local.
         cwid=$(printf "%d\n" "$1") # decimal conversion
         [ -f "/tmp/swallowed-by-$cwid" ] && return 4
-        cpid="$(xdotool getwindowpid "$cwid" 2>/dev/null)" || return 3
+        cpid="$(xdo pid "$cwid")" || return 3
         cname="$(ps -p "$cpid" -o comm=)"
         [ "${blacklist#* $cname }" != "$blacklist" ] && return 0
         process_tree="$(pstree -ATlsp "$cpid" \
                 | sed -n -e 's|(|:|g' -e 's|)||g' \
-                -e 's|---|\n|g' -e 's|-+-|\n|g' -e 's/\n'"$cname"':'"$cpid"'.*//' -e '1p' | tac)"
-        [ "$verbose" -eq 1 ] && \
-        echo "swallow - Scanning process tree of $cname($cpid): $process_tree" | \
-        paste -s -d ' ' # put it all together in a single verbose line
+                -e 's|---|\n|g' -e 's|-+-|\n|g' \
+                -e 's/\n'"$cname"':'"$cpid"'.*//' -e '1p' | tac)"
+        vecho "swallow - Scanning process tree of $cname($cpid): $process_tree"
         for parent in $process_tree; do
                 pname="${parent%%:*}" # ancestor name
                 [ "${swallowable#* $pname }" != "$swallowable" ] \
                         && ppid="${parent##*:}" && break
         done
         [ -z "$ppid" ] && return 1 # nothing swallowable.
-        [ "$verbose" -eq 1 ] && echo "swallow - $pname($ppid) detected as swallowable"
-        pwid="$(xdotool search --limit 1 --onlyvisible --pid "$ppid")"
+        vecho "swallow - $pname($ppid) detected as swallowable"
+        pwid="$(xdo id -p "$ppid" | sed 1q)"
         [ -z "$pwid" ] && return 4
-        [ "$verbose" -eq 1 ] && echo "swallow - $cwid swallowing $pwid using $swallow_command"
-        echo "$swallow_command" | \
+        vecho "swallow - $cwid swallowing $pwid using $swallow_command"
+        xdo activate "$pwid"; echo "$swallow_command" | \
```

Comment:
This one is giving me problems. Why activate before swallowing? It messes up some fullscreen rules (re-focusing takes child program out of fullscreen).

Reply:
It's got here by accident. Changing.

Reply:
Pls push your change.

Reply:
It's to maintain the correct shape in floating WM. (those with dynamic tiling). Eg: bspwm. Sorry I really forgot about it

Reply:
I think it is better to have pre glue hook before swallowing. Ad some wm don't manage the window after unmaping.

Reply:
Fixed it, it was doing nothing. 😖